cmd/agent/app/builder.go

```diff
@@ -128,16 +129,12 @@ func (b *Builder) getReporter(primaryProxy CollectorProxy) reporter.Reporter {
 	return reporter.NewMultiReporter(rep...)
 }
 
-func (b *Builder) publishOpts(mFactory metrics.Factory) {
-	internalFactory := mFactory.Namespace(metrics.NSOptions{Name: "internal"})
+func (b *Builder) publishOpts() {
+	v := expvar.NewInt("jaeger_agent_max_traces")
 	for _, p := range b.Processors {
-		prefix := fmt.Sprintf(processorPrefixFmt, p.Model, p.Protocol)
-		internalFactory.Gauge(metrics.Options{Name: prefix + suffixServerMaxPacketSize}).
-			Update(int64(p.Server.MaxPacketSize))
-		internalFactory.Gauge(metrics.Options{Name: prefix + suffixServerQueueSize}).
-			Update(int64(p.Server.QueueSize))
-		internalFactory.Gauge(metrics.Options{Name: prefix + suffixWorkers}).
-			Update(int64(p.Workers))
+		v.Set(int64(p.Server.MaxPacketSize))
+		v.Set(int64(p.Server.QueueSize))
+		v.Set((int64(p.Workers)))
```

Comment:
so what do you think will happen when you do these three Set ops in a row? How do you expect the users to benefit from that?